filesystem/storage/DBFile.php

```diff
@@ -0,0 +1,457 @@
+<?php
+
+use SilverStripe\Filesystem\ImageManipulation;
+use SilverStripe\Filesystem\Storage\AssetContainer;
+use SilverStripe\Filesystem\Storage\AssetStore;
+
+// Un-comment once https://github.com/silverstripe/silverstripe-framework/pull/4551/ is merged
+// namespace SilverStripe\Filesystem\Storage;
+
+/**
+ * Represents a file reference stored in a database
+ *
+ * @property string $Hash SHA of the file
+ * @property string $Filename Name of the file, including directory
+ * @property string $Variant Variant of the file
+ *
+ * @package framework
+ * @subpackage model
+ */
+class DBFile extends CompositeDBField implements AssetContainer, ShortcodeHandler {
+
+	use ImageManipulation;
+
+	/**
+	 * List of allowed file categories.
+	 *
+	 * {@see File::$app_categories}
+	 *
+	 * @var array
+	 */
+	protected $allowedCategories = array();
+
+	/**
+	 * List of image mime types supported by the image manipulations API
+	 *
+	 * {@see File::app_categories} for matching extensions.
+	 *
+	 * @config
+	 * @var array
+	 */
+	private static $supported_images = array(
+		'image/jpeg',
+		'image/gif',
+		'image/png'
+	);
+
+	/**
+	 * Create a new image manipulation
+	 *
+	 * @param string $name
+	 * @param array|string $allowed List of allowed file categories (not extensions), as per File::$app_categories
+	 */
+	public function __construct($name = null, $allowed = array()) {
+		parent::__construct($name);
+		$this->setAllowedCategories($allowed);
+	}
+
+	/**
+	 * Determine if a valid non-empty image exists behind this asset, which is a format
+	 * compatible with image manipulations
+	 *
+	 * @return boolean
+	 */
+	public function getIsImage() {
```

Comment:
"supported image" is a bit of an ambiguous concept here: It might be displayable via an `<img>` tag, we just don't support processing it with the available `Image_Backend` implementations. So `getFrontendTemplate()` might return empty on images which otherwise would be fine in `<img>`. It's such an edge case that we can probably ignore it though.